tests/tree_construction.rs

```diff
@@ -70,21 +64,32 @@ lazy_static! {
 // #[test_case("template.dat")]
 #[test_case("tests_innerHTML_1.dat")]
 #[test_case("tricky01.dat")]
-#[test_case("webkit01.dat")]
+// #[test_case("webkit01.dat")]
 #[test_case("webkit02.dat")]
 fn tree_construction(filename: &str) {
-    let fixture_file = fixture_from_filename(filename).expect("fixture");
+    let fixture_file =
+        read_fixture_from_path(&get_fixture_root_path().join(filename)).expect("fixture");
+    let mut harness = Harness::new();
 
     for test in fixture_file.tests {
-        if DISABLED.contains(test.data()) {
-            for &scripting_enabled in test.script_modes() {
-                // Check that we don't panic
-                let _ = test.parse(scripting_enabled).expect("problem parsing");
-            }
+        // skip disabled tests
+        if DISABLED_CASES.contains(&test.get_document_as_str()) {
             continue;
         }
 
-        println!("tree construction: {}", test.data());
-        test.assert_valid();
```

Comment:
I think this is a better separation of concerns:  the assert_valid was in a test, but it should not be there. Now it's moved to a regular assert!, where the test simply tells it if it passes or not (is_success).

Reply:
I don't know that I agree with the general principle in this case, but the specific change seems fine.

Reply:
The reason I think it separates this better, is that when doing an assert_valid(), the test case now knows about the specific test system you are using. By just having the test returning an "i'm ok", or "i'm  not ok",.. you can leave the specific way of asserting to the caller.  For instance, in the cargo tests, we use `assert!` for this, but in the html5-parser-tests, we need just the `is_success` to display an `X` or `.`.